hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -590,12 +590,15 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 		// This check can be expensive looking up release image versions
 		// (hopefully they are cached).  Skip if we have already observed for
-		// this generation.
-		if condition == nil || condition.ObservedGeneration != hcluster.Generation {
-			condition := metav1.Condition{
+		// this generation, unless the Status is False, in which case try again
+		if condition == nil {
```

Comment:
how is this additional if needed?
wouldn't the PR just need to add another discriminator factor for the if statement, i.e
`if condition == nil || condition.Status != metav1.ConditionTrue || condition.ObservedGeneration != hcluster.Generation`
?